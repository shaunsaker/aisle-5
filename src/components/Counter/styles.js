import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  spacer: {
    width: styleConstants.dimensions.spacing.horizontal / 2,
  },
});

export default styles;
