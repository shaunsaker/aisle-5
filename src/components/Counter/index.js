import React from 'react';
import PropTypes from 'prop-types';
import { View, Text } from 'react-native';
import Animator from 'react-native-simple-animators';

import styles from './styles';

import IconButton from '../IconButton';

const propTypes = {
  value: PropTypes.number,
  handleChange: PropTypes.func,
};

const defaultProps = {};

const Counter = ({ value, handleChange }) => {
  return (
    <View style={styles.container}>
      <IconButton
        name="remove"
        small
        secondary
        handlePress={() => handleChange(value - 1)}
        disabled={value < 2}
      />

      <Text style={styles.text}>{value}</Text>

      <IconButton name="add" small secondary handlePress={() => handleChange(value + 1)} />
    </View>
  );
};

Counter.propTypes = propTypes;
Counter.defaultProps = defaultProps;

export default Counter;
