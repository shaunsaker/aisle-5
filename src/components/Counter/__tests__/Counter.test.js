import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '..';

describe('Counter', () => {
  const value = 1;

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<Counter value={value} />);

      expect(component).toMatchSnapshot();
    });

    it('renders the default state', () => {
      const component = renderer.create(<Counter value={value + 1} />);

      expect(component).toMatchSnapshot();
    });
  });
});
