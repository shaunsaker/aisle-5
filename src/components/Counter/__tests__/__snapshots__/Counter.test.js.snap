// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter renders renders the default state 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": null,
        "borderRadius": 30,
        "borderWidth": 0,
        "elevation": 6,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 2,
        "width": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#EDE8CA",
              "fontSize": 30,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "color": "#08293A",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "width": 7.5,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": null,
        "borderRadius": 30,
        "borderWidth": 0,
        "elevation": 6,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 2,
        "width": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#EDE8CA",
              "fontSize": 30,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "color": "#08293A",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`Counter renders renders with minimum required props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "paddingHorizontal": 15,
      "paddingVertical": 10,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": null,
        "borderRadius": 30,
        "borderWidth": 0,
        "elevation": 6,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 2,
        "width": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#EDE8CA",
              "fontSize": 30,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "color": "#08293A",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "width": 7.5,
      }
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderColor": null,
        "borderRadius": 30,
        "borderWidth": 0,
        "elevation": 6,
        "height": 30,
        "justifyContent": "center",
        "opacity": 1,
        "shadowColor": "#212121",
        "shadowOffset": Object {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.33,
        "shadowRadius": 2,
        "width": 30,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          Array [
            Object {
              "color": "#EDE8CA",
              "fontSize": 30,
            },
            Object {
              "fontSize": 20,
            },
            Object {
              "color": "#08293A",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;
